quanto/tensor/core.py

```diff
@@ -288,11 +385,18 @@ def __repr__(self):
         return f"QBitsTensor({self._data}, scale={self._scale}, zeropoint={self._zeropoint}, dtype={self.dtype}{autograd_info})"
 
     @classmethod
-    def quantize(cls, base, itype=int4, axis=None):
+    def quantize(cls, base, itype=int4, axis=None, pack=True):
         """Differentiable quantization function"""
-        return AffineQuantizer.apply(base, itype, axis)
+        return AffineQuantizer.apply(base, itype, axis, pack)
 
     def qtensor(self):
+        # Unpack weights for inference
+        if self.packed:
+            # TODO fix this bug
+            self._data = unpack_weights(self._data, int4)
+            # Once packed, no need to unpack them again
+            self.packed = False
```

Comment:
Also this means after doing one inference pass the model is fully unpacked, is that really what we want?